packages/e2e/src/factories.ts

```diff
@@ -273,6 +323,67 @@ export const keyAgentById = memoize(async (accountIndex: number, provider: strin
   return keyManagementFactory.create(provider, params);
 });
 
+/**
+ * Updates the id property of the params object
+ *
+ * @param idx The new id.
+ * @param params the params to be updated.
+ */
+const updateParamsId = function (idx: number, params: any) {
+  params.id = idx;
+  return params;
+};
+
+/**
+ * Create a single wallet instance given the environment variables.
+ *
+ * @param environment The environment variables.
+ * @param stores The wallet store.
+ * @param idx The id of the key agent.
+ */
+export const getWallet = async (environment: any, stores?: storage.WalletStores, idx = 0) => {
```

Comment:
SingleAddressWallet accepts PollingConfig - specifying a lower polling interval would further speed up the tests. Might as well convert these args to an object too.
```suggestion
type GetWalletProps = { environment: any; stores?: storage.WalletStores; idx?: number; pollingInterval?: PollingInterval };
export const getWallet = async ({environment, stores, idx = 0, pollingInterval}: GetWalletProps) => {
```

Reply:
done and squashed :)